### PR TITLE
Restructure docs

### PR DESCRIPTION
**Pull request by tobiscr**
**Description**

Consolidate existing documentation and cleanup.

**Related issue(s)**

### CODE REVIEW

File: docs/user/tutorials/README.md [hunk 1]
```diff
@@ -1,3 +1,48 @@
...
+
+### Prerequisites
+
+Besides the Kyma default modules like Istio and API Gateway, you must enable the following Kyma modules:
+
+* `Application Connector` as it includes the Application Gateway for proxing requests to external systems.
+* `Serverless` is in this example used to run a Function which is sending an HTTP request to the external system.
+
+
+## Initial Setup
```
Comment (grego952) [inline docs/user/tutorials/README.md:33]:
```suggestion
```

File: docs/user/tutorials/README.md [hunk 2]
```diff
@@ -1,3 +1,48 @@
...
+
+
+### Prerequisites
+
+Besides the Kyma default modules like Istio and API Gateway, you must enable the following Kyma modules:
+
+* `Application Connector` as it includes the Application Gateway for proxing requests to external systems.
+* `Serverless` is in this example used to run a Function which is sending an HTTP request to the external system.
+
+
```
Comment (grego952) [inline docs/user/tutorials/README.md:32]:
```suggestion
```

File: docs/user/tutorials/README.md [hunk 3]
```diff
@@ -1,3 +1,48 @@
...
+## Initial Setup
+
+### Create a Kyma Runtime and enable the required Modules
```
Comment (grego952) [inline docs/user/tutorials/README.md:35]:
```suggestion
## Create Kyma Runtime and Enable the Required Modules
```

File: docs/user/tutorials/README.md [hunk 4]
```diff
@@ -1,3 +1,48 @@
...
+
+1. Create a Kyma Runtime either by using the [BTP cockpit](https://help.sap.com/docs/btp/sap-business-technology-platform/create-kyma-environment-instance) or by following the [Kyma Quick Install](https://kyma-project.io/#/02-get-started/01-quick-install) tutorial.
```
Comment (grego952) [inline docs/user/tutorials/README.md:37]:
```suggestion
1. Create Kyma runtime either by using the [SAP BTP cockpit](https://help.sap.com/docs/btp/sap-business-technology-platform/create-kyma-environment-instance) or by following the [Kyma Quick Install](https://kyma-project.io/#/02-get-started/01-quick-install) tutorial.
```

File: docs/user/tutorials/README.md [hunk 5]
```diff
@@ -1,3 +1,48 @@
...
+
+2. Add following Kyma modules: `Application Connector` and `Serverless`. If you created the Kyma runtime via BTP, follow this tutorial for [Adding and Deleting a Kyma Module](https://help.sap.com/docs/btp/sap-business-technology-platform/enable-and-disable-kyma-module?#add-and-delete-a-kyma-module-using-kyma-dashboard). Alterantively, continue with the steps described in the [Kyma Quick Install](https://kyma-project.io/#/02-get-started/01-quick-install?id=steps) tutorial.
```
Comment (grego952) [inline docs/user/tutorials/README.md:39]:
```suggestion
2. Add the Application Connector and Serverless modules. If you created Kyma runtime using SAP BTP cockpit, follow this tutorial for [Adding and Deleting a Kyma Module](https://help.sap.com/docs/btp/sap-business-technology-platform/enable-and-disable-kyma-module?#add-and-delete-a-kyma-module-using-kyma-dashboard). Alternatively, follow the steps described in the [Kyma Quick Install](https://kyma-project.io/#/02-get-started/01-quick-install?id=steps) tutorial.
```

File: docs/user/tutorials/README.md [hunk 6]
```diff
@@ -1,3 +1,48 @@
...
+
```
Comment (grego952) [inline docs/user/tutorials/README.md:40]:
```suggestion
```

File: docs/user/tutorials/README.md [hunk 7]
```diff
@@ -1,3 +1,48 @@
...
+
+## Steps
```
Comment (grego952) [inline docs/user/tutorials/README.md:42]:
```suggestion
## Integrate an External System with Kyma Runtime
```

File: docs/user/tutorials/README.md [hunk 8]
```diff
@@ -1,3 +1,48 @@
...
+
+1. [Create an Application Custom Resource](./01-10-create-application.md): the Application CR represents an external system and contains all information about exposed endpoints and their security configuration etc.
```
Comment (grego952) [inline docs/user/tutorials/README.md:44]:
```suggestion
1. [Create an Application custom resource (CR)](./01-10-create-application.md). The Application CR represents an external system and contains all information about exposed endpoints and their security configuration etc.
```

File: docs/user/tutorials/README.md [hunk 9]
```diff
@@ -1,3 +1,48 @@
...
+2. [Register a service for the external system](./01-20-register-manage-services.md) in the Application CR: the service is an abstraction of the external system. Kyma workloads can send their requests to the the service URL and the Application Gateway will proxy these requests to the external system and handling all security related aspects transparently (e.g. establishing a trusted connection or authentication).
```
Comment (grego952) [inline docs/user/tutorials/README.md:45]:
```suggestion
2. [Register a service for the external system](./01-20-register-manage-services.md) in the Application CR. The service is an abstraction of the external system. Kyma workloads can send their requests to the service URL, and the Application Gateway proxies these requests to the external system and handles all security-related aspects transparently (for example, establishing a trusted connection or authentication).
```

File: docs/user/tutorials/README.md [hunk 10]
```diff
@@ -1,3 +1,48 @@
...
+    * [Register a secured API](./01-30-register-secured-api.md):  many different authentication methods are supported by the Application Connector, you can find in this step for each method an example.
```
Comment (grego952) [inline docs/user/tutorials/README.md:46]:
```suggestion
    * [Register a secured API](./01-30-register-secured-api.md). The Application Connector module supports many different authentication methods. In this step, you can find an example for each of them.
```

File: docs/user/tutorials/README.md [hunk 11]
```diff
@@ -1,3 +1,48 @@
...
+    * [Disable TLS certificate validation](./01-50-disable-tls-certificate-verification.md): for testing purposes, it can be helpful to disable the TLS certificate validation. The Application  Connector is per default validating TLS certificates when estalishing a secure connection.
```
Comment (grego952) [inline docs/user/tutorials/README.md:47]:
```suggestion
    * [Disable TLS certificate validation](./01-50-disable-tls-certificate-verification.md). For testing purposes, disabling the TLS certificate validation can be helpful. By default, the Application Connector module validates TLS certificates when establishing a secure connection.
```

File: docs/user/tutorials/README.md [hunk 12]
```diff
@@ -1,3 +1,48 @@
...
+3. [Call the registered API](./01-40-call-registered-service-from-kyma.md): this step shows how to call an external system by using a Kyma function.
```
Comment (grego952) [inline docs/user/tutorials/README.md:48]:
```suggestion
3. [Call the registered API](./01-40-call-registered-service-from-kyma.md). In this step, you can learn how to call an external system using a Kyma Function.
```

File: docs/user/tutorials/README.md [hunk 13]
```diff
@@ -1,3 +1,48 @@
...
+
+## Introduction
+
+The document describes the steps for connecting an external system (e.g. HTTPBin) with a Kyma runtime.
+
+In this example, Kyma will send an authenticated requests to an external service.
+
+
+### Prerequisites
+
```
Comment (grego952) [inline docs/user/tutorials/README.md:26]: Maybe let's move everything from the **Create a Kyma Runtime and enable the required Modules** subsection to the Prerequisites? This way, we can get rid of the whole Initial Setup section.

File: docs/user/00-10-architecture.md [hunk 14]
```diff
@@ -0,0 +1,115 @@
+
+# Architecture
+
+In the architectural diagram below are all components of the Application Connector Module included. It describes also the workflow how the different components integrate an external system into Kyma.
+
+
+## Components
+
+To get more details about a particular component, please follow the link.
```
[the remark this answers is not in the document]
Comment (tobiscr) [reply docs/user/00-10-architecture.md]: Dropped the sentence - not needed.

File: docs/user/tutorials/README.md (hunk 13 above)
Comment (tobiscr) [reply docs/user/tutorials/README.md:24]: Adjusted :)

File: docs/user/tutorials/README.md [hunk 15]
```diff
@@ -1,3 +1,48 @@
-# Application Connectivity - Tutorials
+<span class="fd-avatar--thumbnail fd-avatar fd-avatar--40 fd-avatar--accent-color-10" style="background-image: url('https://avatars.githubusercontent.com/tobiscr')" role="img"></span> [Tobias Schuhmacher](https://github.com/tobiscr) - Product Owner of Team Framefrog, Kyma runtime
 
-Browse the tutorials for Application Connectivity to learn how to use it step-by-step in different scenarios.
+Last update: Nov 2024
+
+# Integrate external system with Kyma
+
+## Table of Contents
```
[the remark this answers is not in the document]
Comment (tobiscr) [reply docs/user/tutorials/README.md]: Sure, dropped

File: docs/user/resources/04-10-application.md [hunk 16]
```diff
@@ -44,16 +46,16 @@ This table lists all the possible parameters of a given resource together with t
...
-| **spec.services.entries** | Yes | Contains the information about the APIs and events that the service offered by the Application provides. In the [Compass mode](../README.md), it's provided by Runtime Agent. In the [standalone mode](../README.md), you must provide it yourself. |
-| **spec.services.entries.type** | Yes | Specifies the entry type: `API` or `Events`. In the [Compass mode](../README.md), it's provided by Runtime Agent. In the [standalone mode](../README.md), you must provide it yourself. |
-| **spec.services.entries.centralGatewayUrl** | No | Specifies the URL of Application Gateway. An internal address is resolvable only within the cluster. This field is required for the API entry type. In the [Compass mode](../README.md), it's provided by Runtime Agent. In the [standalone mode](../README.md), you must provide it yourself. |
+| **spec.services.providerDisplayName** | Yes | Specifies a human-readable name of the Application service provider. |Yes|Yes|
```
[the remark this answers is not in the document]
Comment (tobiscr) [reply docs/user/resources/04-10-application.md]: oh sorry, that's not required. I've dropped these two YES

File: docs/user/README.md [hunk 17]
```diff
@@ -1,69 +1,56 @@
...
+If an external system is registered for Kyma runtime in SAP BTP's UCL, the Application Connector module automatically configures it and can send requests to Kyma workloads. The Application Connector module includes [`Runtime Agent`](./technical-reference/runtime-agent/README.md) and acts as a client of the UCL backend. It automatically retrieves the configuration of each external system and integrates it into Kyma.
 
-## Useful Links
 
-If you want to perform some simple and more advanced tasks, check the [Application Connectivity tutorials](tutorials/README.md).
+#### Manually
 
-To learn more about the architecture, the configuration parameters, and any other references, visit [technical reference](technical-reference/README.md).
+It is always possible to integrate any external system into Kyma by applying the configuration by hand. The steps for configuring and integrating a new external system in your Kyma runtime are described in the [Integrate an External System with Kyma](tutorials/01-11-integrate-external-system.md).
```
Comment (grego952) [inline docs/user/README.md:55]:
```suggestion
It is always possible to integrate any external system into Kyma by applying the configuration by hand. The steps for configuring and integrating a new external system in your Kyma runtime are described in the [Integrate an External System with Kyma](tutorials/01-00-integrate-external-system.md).
```

File: docs/user/technical-reference/runtime-agent/03-10-ucl-connection.md [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+The connection status is preserved in the [Connection](../../resources/04-20-compassconnection.md) custom resource (CR). This CR also stores the Connector URL and the Director URL.
+
+> [!NOTE]
+> Mind that UCL was previously called Compass. For historical reasons, the Connection CR is still called `CompassConnection`.
+
+## Reconnecting
+
+If the connection with UCL fails, Runtime Agent keeps trying to connect with the token from the Secret. If the connection is established successfully, Runtime Agent ignores the Secret until the connection is lost.
+
+To see how to reconnect Runtime Agent with UCL, see the [tutorial](./tutorials/01-100-reconnect-runtime-agent-with-compass.md).
```
Comment (grego952) [inline docs/user/technical-reference/runtime-agent/03-10-ucl-connection.md:29]:
```suggestion
To see how to reconnect Runtime Agent with UCL, see this [tutorial](./tutorials/01-100-reconnect-runtime-agent-with-compass.md).
```

File: docs/user/technical-reference/runtime-agent/03-20-configuring-runtime.md [hunk 19]
```diff
@@ -0,0 +1,39 @@
+# Configuring the Runtime
+
+Runtime Agent periodically requests for the configuration of Kyma runtime from Unified Customer Landscape (UCL).
+
+To fetch the Kyma runtime configuration, Runtime Agent calls the [`applicationsForRuntime`](https://github.com/kyma-incubator/compass/blob/master/components/director/pkg/graphql/schema.graphql) query offered by the component called UCL Director.
+
+The response for the query contains Applications (an `Application` represents an external system) assigned for Kyma runtime.
```
Comment (grego952) [inline docs/user/technical-reference/runtime-agent/03-20-configuring-runtime.md:7]:
```suggestion
The response for the query contains Applications (an Application represents an external system) assigned for Kyma runtime.
```

File: docs/user/technical-reference/runtime-agent/03-20-configuring-runtime.md [hunk 20]
```diff
@@ -0,0 +1,39 @@
...
+
+Each `Application` contains credentials that are only valid and unique for the querying Kyma runtime (another requestor cannot use them). Runtime Agent stores the credentials in a Secrets which will be used by Application Gateway for establishing a trusted outbound communication to an external system.
```
Comment (grego952) [inline docs/user/technical-reference/runtime-agent/03-20-configuring-runtime.md:9]:
```suggestion
Each Application contains credentials that are only valid and unique for querying Kyma runtime (another requestor cannot use them). Runtime Agent stores the credentials in Secrets, which are used by Application Gateway to establish a trusted outbound communication to an external system.
```

File: docs/user/technical-reference/runtime-agent/03-20-configuring-runtime.md [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+
+This data mapping shows how the retrieved configuration of an `Application` from UCL Director is stored in Kyma runtime:
```
Comment (grego952) [inline docs/user/technical-reference/runtime-agent/03-20-configuring-runtime.md:11]:
```suggestion
This data mapping shows how the retrieved configuration of an Application from UCL Director is stored in Kyma runtime:
```

File: docs/user/technical-reference/runtime-agent/03-20-configuring-runtime.md [hunk 22]
```diff
@@ -0,0 +1,39 @@
...
+
+| **UCL Director**    | **Kyma Runtime**                    |
+|---------------------------|-------------------------------|
+| Application               | Application CR                |
+| API Bundle                | Service in the Application CR |
+| API Definition            | Entry under the service       |
+
+
+## Application Name
+The name of the `Application` is used as a key within the Application Connector module and has special requirements:
```
Comment (grego952) [inline docs/user/technical-reference/runtime-agent/03-20-configuring-runtime.md:21]:
```suggestion
The name of the Application is used as a key within the Application Connector module and has the following special requirements:
```

File: docs/user/technical-reference/runtime-agent/README.md [hunk 23]
```diff
@@ -0,0 +1,50 @@
...
+
+### Workflow
+
+1. The Runtime Agent fetches the certificate from the UCL Connector to initialize the connection with the UCL system.
+
+2. The certificate and key for the UCL Connector and the UCL Director are stored in a Kubernetes Secret.
+
+3. The agent synchronizes the Kyma-integrated external systems with the UCL Director by:
+
+    * Fetching new external systems from UCL Director and creating them in Kyma runtime
```
Comment (grego952) [inline docs/user/technical-reference/runtime-agent/README.md:35]:
```suggestion
    * Fetching new external systems from the UCL Director and creating them in Kyma runtime
```

File: docs/user/technical-reference/runtime-agent/README.md [hunk 24]
```diff
@@ -0,0 +1,50 @@
...
+
+    * Removing from Kyma runtime the external systems that no longer exist in the UCL Director.
+
+4. It reports the Event Gateway URL and the Dashboard URL of Kyma runtime to UCL Director. These URLs are also displayed in the UCL UI.
```
Comment (grego952) [inline docs/user/technical-reference/runtime-agent/README.md:39]:
```suggestion
4. It reports the Event Gateway URL and the Dashboard URL of Kyma runtime to the UCL Director. These URLs are also displayed in the UCL UI.
```

File: docs/user/tutorials/01-00-integrate-external-system.md [hunk 25]
```diff
@@ -0,0 +1,26 @@
...
+
+## Introduction
+
+The document describes the steps needed to connect an external system (for example, HTTPBin) with a Kyma runtime.
+
+In this example, Kyma sends authenticated requests to an external service.
+
+## Prerequisites
+
+Besides the Kyma default modules like Istio and API Gateway, you must enable the following Kyma modules:
```
Comment (grego952) [inline docs/user/tutorials/01-00-integrate-external-system.md:11]:
```suggestion
```

File: docs/user/tutorials/01-00-integrate-external-system.md [hunk 26]
```diff
@@ -0,0 +1,26 @@
...
+
+1. Create Kyma runtime either by using the [SAP BTP cockpit](https://help.sap.com/docs/btp/sap-business-technology-platform/create-kyma-environment-instance) or by following the [Kyma Quick Install](https://kyma-project.io/#/02-get-started/01-quick-install) tutorial.
+
+2. Add following Kyma modules:
```
Comment (grego952) [inline docs/user/tutorials/01-00-integrate-external-system.md:15]:
```suggestion
2. Besides the Kyma default modules like Istio and API Gateway, you must enable the following Kyma modules:
```

File: docs/user/tutorials/01-00-integrate-external-system.md [hunk 27]
```diff
@@ -0,0 +1,26 @@
...
+    * `Application Connector`: it includes the Application Gateway for proxing requests to external systems.
+    * `Serverless`: used to run a Function which is sending an HTTP request to the external system.
```
Comment (grego952) [inline docs/user/tutorials/01-00-integrate-external-system.md:17]:
```suggestion
    * `Serverless`: used to run a Function which is sending an HTTP request to an external system.
```

File: docs/user/tutorials/01-00-integrate-external-system.md [hunk 28]
```diff
@@ -0,0 +1,26 @@
...
+    If you created the Kyma runtime via BTP, follow this tutorial for [Adding and Deleting a Kyma Module](https://help.sap.com/docs/btp/sap-business-technology-platform/enable-and-disable-kyma-module?#add-and-delete-a-kyma-module-using-kyma-dashboard). Alterantively, continue with the steps described in the [Kyma Quick Install](https://kyma-project.io/#/02-get-started/01-quick-install?id=steps) tutorial.
```
Comment (grego952) [inline docs/user/tutorials/01-00-integrate-external-system.md:18]:
```suggestion
    If you created Kyma runtime using SAP BTP cockpit, follow this tutorial for [Adding and Deleting a Kyma Module](https://help.sap.com/docs/btp/sap-business-technology-platform/enable-and-disable-kyma-module?#add-and-delete-a-kyma-module-using-kyma-dashboard). Alternatively, continue with the steps described in the [Kyma Quick Install](https://kyma-project.io/#/02-get-started/01-quick-install?id=steps) tutorial.
```

File: docs/user/00-10-architecture.md [hunk 29]
```diff
@@ -0,0 +1,108 @@
+# Architecture
+
+The following architectural diagram includes all components of the Application Connector module and describes the workflow for integrating an external system into Kyma using the different components.
+
+## Components
+
+|Component|Purpose|
+|--|--|
+|External application|An external system that interacts with your Kyma workload or is called by it.|
+|Unified Customer Landscape (UCL)|The UCL system implements the BTP Extensibility Concept and administrates system formations.|
```
Comment (grego952) [inline docs/user/00-10-architecture.md:10]:
```suggestion
|Unified Customer Landscape (UCL)|The UCL system implements the SAP Extensibility concept and administrates system formations.|
```

File: docs/user/README.md [hunk 30]
```diff
@@ -2,68 +2,39 @@
...
-- Provides certificate handling for the [Eventing](https://kyma-project.io/#/eventing-manager/user/README) flow in the Compass scenario (mode)
-- Manages secure access to external systems
-- Provides monitoring and tracing capabilities to facilitate operational aspects
-
-Depending on your use case, Application Connectivity works in one of the two modes:
-
-- **Standalone mode** (default) - a standalone mode where Kyma is not connected to [Compass](https://github.com/kyma-incubator/compass)
-- **Compass mode** - using [Runtime Agent](00-30-runtime-agent-overview.md) and integration with [Compass](https://github.com/kyma-incubator/compass) to automate connection and registration of services using mTLS certificates
+* Smooth and loosely coupled integration of external systems with Kyma workloads using [Kyma Eventing](https://kyma-project.io/#/eventing-manager/user/README)
+* Easy consumption of SAP BTP services by supporting the SAP BTP Extensibility approach
```
Comment (grego952) [inline docs/user/README.md:8]:
```suggestion
* Easy consumption of SAP BTP services by supporting the SAP Extensibility approach
```

File: docs/user/README.md [hunk 31]
```diff
@@ -2,68 +2,39 @@
...
-In addition to authentication methods, Application Connector supports Cross-Site Request Forgery (CSRF) Tokens.
+## Features
 
-Application Connector supports any API that adheres to the REST principles and is available over the HTTP protocol.
+The Application Connector module provides the following features:
 
-## Application Connector Manager
+* Easy installation of Kyma's Application Connectivity capabilities by enabling the Application Connector module in your Kyma runtime.
+* Simple configuration using Kubernetes CRs and easy management with Kyma dashboard.
+* Full integration of BTP's UCL service, which implements the SAP BTP Extensibility concept. This allows for the automated integration of external systems registered in the UCL service.
```
Comment (grego952) [inline docs/user/README.md:27]:
```suggestion
* Full integration of SAP BTP's UCL service, which implements the SAP Extensibility concept. This allows for the automated integration of external systems registered in the UCL service.
```

File: docs/user/resources/04-10-application.md [hunk 32]
```diff
@@ -1,6 +1,8 @@
-# Application
+# `Application` Custom Resource
 
-The `applications.applicationconnector.kyma-project.io` CustomResourceDefinition (CRD) is a detailed description of the kind of data and the format used to register the external solution connected to Kyma represented as an Application. The `Application` custom resource (CR) defines the APIs that the Application offers. To get the up-to-date CRD and show the output in the YAML format, run this command:
+The `applications.applicationconnector.kyma-project.io` CustomResourceDefinition (CRD) describes a kind of data and the format used to register an external system connected to Kyma runtime. 
+
+An instance of the CRD is called a custom resource (CR). The custom resource of this particular CRD is of kind  `Application`. The `Application` CR stores all data related to the APIs exposed by an external system. To get the up-to-date CRD and show the output in the YAML format, run this command:
```
Comment (grego952) [inline docs/user/resources/04-10-application.md:5]:
```suggestion
An instance of the CRD is called a custom resource (CR). The CR of this particular CRD is of kind  `Application`. The `Application` CR stores all data related to the APIs exposed by an external system. To get the up-to-date CRD and show the output in the YAML format, run this command:
```

File: docs/user/technical-reference/02-10-istio-gateway.md [hunk 33]
```diff
@@ -0,0 +1,22 @@
...
+
+## DNS Name
+
+The DNS name of Ingress Gateway is cluster-dependent. For the managed Kyma clusters, it follows the `gateway.{cluster-dns}` format.
+
+## Security
+
+### Client Certificates
+
+For external systems automatically integrated by UCL, the Application Connector module uses the mutual TLS protocol (mTLS) with client authentication enabled. As a result, anyone attempting to connect to the Application Connector module must present a valid client certificate dedicated to a specific Application. This way, the traffic is fully encrypted, and the client has a valid identity.
```
Comment (grego952) [inline docs/user/technical-reference/02-10-istio-gateway.md:17]:
```suggestion
For external systems automatically integrated by Unified Customer Landscape (UCL), the Application Connector module uses the mutual TLS protocol (mTLS) with client authentication enabled. As a result, anyone attempting to connect to the Application Connector module must present a valid client certificate dedicated to a specific Application. This way, the traffic is fully encrypted, and the client has a valid identity.
```

File: docs/user/technical-reference/02-20-application-gateway.md [hunk 34]
```diff
@@ -0,0 +1,99 @@
+# Application Gateway
+
+Application Gateway is an intermediary component between a Kyma workload and an external system. It proxies outgoing requests from Kyma to the external system API based on the configuration stored in the [`Application`](../resources/04-10-application.md) CR.
```
Comment (grego952) [inline docs/user/technical-reference/02-20-application-gateway.md:3]:
```suggestion
Application Gateway is an intermediary component between a Kyma workload and an external system. It proxies outgoing requests from Kyma to the external system API based on the configuration stored in the [Application](../resources/04-10-application.md) CR.
```

File: docs/user/technical-reference/02-20-application-gateway.md [hunk 35]
```diff
@@ -0,0 +1,99 @@
...
+
+1. A Kyma workload calls Application Gateway which extracts the application name and the service name from the URL path.
+2. Using the extracted application name, the gateway finds the corresponding Application CR and obtains the information about the registered external API, such as the API URL and security credentials.
+3. Application Gateway gets a token from the OAuth server.
+4. This step is optional and is valid only for the API which was registered with a CSRF token enabled: Application Gateway gets a CSRF token from the endpoint exposed by the upstream service.
+5. Application Gateway calls the target API.
+
+## Request Proxying
+
+Application Gateway proxies requests from Kyma workloads to external system APIs based on the configuration stored in the [`Application` CR](../resources/04-10-application.md) and Kubernetes Secrets.
```
Comment (grego952) [inline docs/user/technical-reference/02-20-application-gateway.md:19]:
```suggestion
Application Gateway proxies the requests from Kyma workloads to external system APIs based on the configuration stored in the [Application CR](../resources/04-10-application.md) and Kubernetes Secrets.
```

File: docs/user/technical-reference/runtime-agent/03-20-configuring-runtime.md [hunk 36]
```diff
@@ -0,0 +1,40 @@
+# Configuring the Runtime
+
+Runtime Agent periodically requests for the configuration of Kyma runtime from Unified Customer Landscape (UCL).
+
+To fetch the Kyma runtime configuration, Runtime Agent calls the [`applicationsForRuntime`](https://github.com/kyma-incubator/compass/blob/master/components/director/pkg/graphql/schema.graphql) query offered by the component called UCL Director.
```
Comment (grego952) [inline docs/user/technical-reference/runtime-agent/03-20-configuring-runtime.md:5]:
```suggestion
To fetch the Kyma runtime configuration, Runtime Agent calls the [`applicationsForRuntime`](https://github.com/kyma-incubator/compass/blob/master/components/director/pkg/graphql/schema.graphql) query offered by the component called the UCL Director.
```

File: docs/user/technical-reference/runtime-agent/03-20-configuring-runtime.md [hunk 37]
```diff
@@ -0,0 +1,40 @@
...
+
+The response for the query contains Applications (an Application represents an external system) assigned for Kyma runtime.
+
+Each Application contains credentials that are only valid and unique for querying Kyma runtime (another requestor cannot use them). Runtime Agent stores the credentials in Secrets, which are used by Application Gateway to establish a trusted outbound communication to an external system.
+
+This data mapping shows how the retrieved configuration of an Application from UCL Director is stored in Kyma runtime:
```
Comment (grego952) [inline docs/user/technical-reference/runtime-agent/03-20-configuring-runtime.md:11]:
```suggestion
This data mapping shows how the retrieved configuration of an Application from the UCL Director is stored in Kyma runtime:
```

File: docs/user/technical-reference/runtime-agent/03-20-configuring-runtime.md [hunk 38]
```diff
@@ -0,0 +1,40 @@
...
+
+The name of the Application is used as a key within the Application Connector module and has the following special requirements:
+
+### Uniqueness
+
+The names of Applications assigned to the Runtime must be unique in Kyma Runtime. If they are not unique, the synchronization fails.
+
+### Normalization of Application Names
+
+Runtime Agent can normalize the names of Applications fetched from UCL Director by converting them to lowercase and removing special characters and spaces.
```
Comment (grego952) [inline docs/user/technical-reference/runtime-agent/03-20-configuring-runtime.md:29]:
```suggestion
Runtime Agent can normalize the names of Applications fetched from the UCL Director by converting them to lowercase and removing special characters and spaces.
```

File: docs/user/technical-reference/runtime-agent/README.md [hunk 39]
```diff
@@ -0,0 +1,44 @@
+# Runtime Agent
+
+Runtime Agent acts as a client for UCL (Unified Customer Landscape). Its purpose is to integrate external systems, registered for Kyma in UCL, into Kyma runtime.
+
+Its responsibilities are:
+
+- Regularly synchronizing external systems (so-called `Applications` in UCL) over UCL Director by fetching new Applications from the Director and integrating them into Kyma or removing those that no longer exist.
```
Comment (grego952) [inline docs/user/technical-reference/runtime-agent/README.md:7]:
```suggestion
- Regularly synchronizing external systems (so-called `Applications` in UCL) over the UCL Director by fetching new Applications from the Director and integrating them into Kyma or removing those that no longer exist.
```

File: docs/user/technical-reference/runtime-agent/README.md [hunk 40]
```diff
@@ -0,0 +1,44 @@
...
+
+![Runtime Agent architecture](../../assets/agent-architecture.png)
+
+### Components
+
+|Component|Purpose|
+|--|--|
+|Runtime Agent|Client of the UCL system, which integrates and synchronizes external applications with Kyma runtime. It also ensures trusted communication with the UCL services by exchanging certificates.|
+|[UCL Connector](https://github.com/kyma-incubator/compass/blob/main/docs/connector/02-01-connector-service.md)|Connector establishes trust between external systems and runtimes.|
+|[UCL Director](https://github.com/kyma-incubator/compass/blob/master/docs/compass/02-01-components.md#director)|UCL Director handles the process of managing external systems and runtimes.|
```
Comment (grego952) [inline docs/user/technical-reference/runtime-agent/README.md:24]:
```suggestion
|[UCL Director](https://github.com/kyma-incubator/compass/blob/master/docs/compass/02-01-components.md#director)|The UCL Director handles the process of managing external systems and runtimes.|
```

File: docs/user/technical-reference/runtime-agent/tutorials/01-90-configure-runtime-agent-with-compass.md [hunk 41]
```diff
@@ -0,0 +1,33 @@
...
+## Prerequisites
+
+- [UCL](https://github.com/kyma-incubator/compass) (previously called Compass)
+- Kyma runtime connected to UCL and the Runtime ID
+- [Connector URL](01-60-establish-secure-connection-with-compass.md)
+- One-time token from the Connector
+- Tenant ID
+
+> [!NOTE]
+> Learn also about the [parameters required](../03-10-ucl-connection.md) to initialize the connection between Runtime Agent and Compass.
```
Comment (grego952) [inline docs/user/technical-reference/runtime-agent/tutorials/01-90-configure-runtime-agent-with-compass.md:14]:
```suggestion
> Learn also about the [parameters required](../03-10-ucl-connection.md) to initialize the connection between Runtime Agent and UCL.
```

File: docs/user/tutorials/01-00-integrate-external-system.md [hunk 42]
```diff
@@ -0,0 +1,23 @@
+# Integrate an External System with Kyma
+
+## Introduction
+
+The document describes the steps needed to connect an external system (for example, HTTPBin) with a Kyma runtime.
```
Comment (grego952) [inline docs/user/tutorials/01-00-integrate-external-system.md:5]:
```suggestion
The document describes the steps needed to connect an external system (for example, HTTPBin) with Kyma runtime.
```

File: docs/user/tutorials/01-00-integrate-external-system.md [hunk 43]
```diff
@@ -0,0 +1,23 @@
...
+
+In this example, Kyma sends authenticated requests to an external service.
+
+## Prerequisites
+
+1. Create Kyma runtime either by using the [SAP BTP cockpit](https://help.sap.com/docs/btp/sap-business-technology-platform/create-kyma-environment-instance) or by following the [Kyma Quick Install](https://kyma-project.io/#/02-get-started/01-quick-install) tutorial.
+2. Besides the Kyma default modules like Istio and API Gateway, you must enable the following Kyma modules:
+    * `Application Connector`: it includes the Application Gateway for proxing requests to external systems.
```
Comment (grego952) [inline docs/user/tutorials/01-00-integrate-external-system.md:13]:
```suggestion
    * Application Connector: it includes the Application Gateway for proxying requests to external systems.
```

**Review by grego952 — APPROVED** [review]

**Pull request merged**